### DISCUSSION TITLE
gatsby-transformer-remark - add additional helpful fields to MarkdownRemark nodes from parent `File` node

### DISCUSSION THREAD

**Discussion by kimbaudi** [gatsbyjs/gatsby#30393] [Ideas / Feature Requests]
Currently, `MarkdownRemark` nodes created from `gatsby-transformer-remark` plugin provides `fileAbsolutePath` field and I use it to filter markdown nodes

```gql
{
  allBlogPost: allMarkdownRemark(
    filter: { fileAbsolutePath: { glob: "**/blog-posts/**" } }
  ) {
    nodes {
      fileAbsolutePath
    }
  }
}
```

But if `gatsby-transformer-remark` plugin also provided `sourceInstanceName`, `relativeDirectory`, `name` fields from its parent `File` node, then my graphql query to filter markdown nodes could be simplified to this:

```gql
{
  allBlogPost: allMarkdownRemark(
    filter: { fileSourceInstanceName: { eq: "blog-posts" } }
  ) {
    nodes {
      fileSourceInstanceName
    }
  }
}
```

---

It would be awesome if `gatsby-transformer-remark` could provide the following nodes:

```gql
allMarkdownRemark {
  nodes {
    fileAbsolutePath
    fileSourceInstanceName
    fileRelativeDirectory
    fileName
  }
}
```

or better yet:

```gql
allMarkdownRemark {
  nodes {
    file {
      absolutePath
      sourceInstanceName
      relativeDirectory
      name
    }
  }
}
```

**Comment by whitphx** [comment]
For those who want to do it before the plugin supports it,
[Gatsbyで2種類のマークダウンファイルの区別する方法, Takumon Blog](https://takumon.com/how-to-distinct-2-kinds-of-markdown-in-gatsby) explains the solution. (It's in Japanese so use some translator)
